### PR TITLE
Fix for Constructor injection when using OpenApi annotations in JAX-Rs resource

### LINKED ISSUE #11313: Constructor injection in JAX-RS Resource using OpenApi Annotations causes RuntimeException
**Describe the bug**
JAX-RS resources which have constructor injection and contains openapi annotations causes RuntimeException, this issued is related to #10430

**Expected behavior**
Constructor injection also work like the field injection.

**Actual behavior**
On startup:
Caused by: java.lang.RuntimeException: RESTEASY003190: Could not find constructor for class: xx.xxx.IncidentResource

**To Reproduce**
Is it supposed to be fixed on 1.6.1.Final? I'm using this version and still having the issue, [here a sample](https://github.com/mageddo/poc/tree/1128295/quarkus-swagger-issue)

Explicitly add a NonArg constructor fixes the problem, I expect Quarkus to solve it for me though

```java
@Tag(name = "hello-samples")
@Path("/hello")
public class FruitResource {

    HelloService helloService;

    public FruitResource() {
    }

    @Inject
    public FruitResource(HelloService helloService) {
        this.helloService = helloService;
    }
....
```

### PR DESCRIPTION
**Pull request by vietk**
Fix #11313

### CODE REVIEW

File: extensions/resteasy-server-common/spi/src/main/java/io/quarkus/resteasy/server/common/spi/FriendlyJaxRsAnnotationPrefixBuildItem.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package io.quarkus.resteasy.server.common.spi;
+
+import org.jboss.jandex.DotName;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+public final class FriendlyJaxRsAnnotationPrefixBuildItem extends MultiBuildItem {
+
+    private final DotName annotationClass;
```
Comment (gsmet) [inline extensions/resteasy-server-common/spi/src/main/java/io/quarkus/resteasy/server/common/spi/FriendlyJaxRsAnnotationPrefixBuildItem.java:9]: We usually use plain strings for the class names in the build items.

File: extensions/resteasy-server-common/spi/src/main/java/io/quarkus/resteasy/server/common/spi/FriendlyJaxRsAnnotationPrefixBuildItem.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package io.quarkus.resteasy.server.common.spi;
+
+import org.jboss.jandex.DotName;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+public final class FriendlyJaxRsAnnotationPrefixBuildItem extends MultiBuildItem {
```
Comment (gsmet) [inline extensions/resteasy-server-common/spi/src/main/java/io/quarkus/resteasy/server/common/spi/FriendlyJaxRsAnnotationPrefixBuildItem.java:7]: This needs proper Javadoc. Not entirely sure about the name.

File: extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java [hunk 3]
```diff
@@ -159,6 +160,22 @@ OpenApiFilteredIndexViewBuildItem smallryeOpenApiIndex(CombinedIndexBuildItem co
...
+        prefixes.add(new FriendlyJaxRsAnnotationPrefixBuildItem(
+                DotName.createSimple("org.eclipse.microprofile.openapi.annotations.tags")));
+        prefixes.add(new FriendlyJaxRsAnnotationPrefixBuildItem(
+                DotName.createSimple("org.eclipse.microprofile.openapi.annotations.servers")));
+        prefixes.add(new FriendlyJaxRsAnnotationPrefixBuildItem(
+                DotName.createSimple("org.eclipse.microprofile.openapi.annotations.security")));
+        prefixes.add(new FriendlyJaxRsAnnotationPrefixBuildItem(
+                DotName.createSimple("org.eclipse.microprofile.openapi.annotations.media")));
+        prefixes.add(new FriendlyJaxRsAnnotationPrefixBuildItem(
+                DotName.createSimple("org.eclipse.microprofile.openapi.annotations.extensions")));
```
Comment (gsmet) [inline extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java:175]: Why not adding the whole `org.eclipse.microprofile.openapi.annotations` package?

**Review by gsmet — CHANGES_REQUESTED** [review]
Nice work. I added some suggestions.

File: extensions/resteasy-server-common/spi/src/main/java/io/quarkus/resteasy/server/common/spi/FriendlyJaxRsAnnotationPrefixBuildItem.java (hunk 2 above)
Comment (vietk) [reply extensions/resteasy-server-common/spi/src/main/java/io/quarkus/resteasy/server/common/spi/FriendlyJaxRsAnnotationPrefixBuildItem.java]: I added some

File: extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java (hunk 3 above)
Comment (vietk) [reply extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java]: The implementation of the detection of "allowed" annotation was not able to detect the root package of an annotation, only the package of the annotation itself, that's why I included all package of every OpenApi annotations.
I improved the detection to declare the root package of a set of annotations like OpenAPI

File: extensions/resteasy-server-common/spi/src/main/java/io/quarkus/resteasy/server/common/spi/FriendlyJaxRsAnnotationPrefixBuildItem.java (hunk 1 above)
Comment (vietk) [reply extensions/resteasy-server-common/spi/src/main/java/io/quarkus/resteasy/server/common/spi/FriendlyJaxRsAnnotationPrefixBuildItem.java]: OK, I used plain strings

**Review by gsmet — APPROVED** [review]

**Pull request merged**